lib/ansible/modules/dnf.py

```diff
@@ -1125,25 +1106,11 @@ def ensure(self):
                         # Environment is already uninstalled.
                         pass
 
-                installed = self.base.sack.query().installed()
                 for pkg_spec in pkg_specs:
-                    # short-circuit installed check for wildcard matching
-                    if '*' in pkg_spec:
-                        try:
-                            self.base.remove(pkg_spec)
-                        except dnf.exceptions.MarkingError as e:
-                            is_failure, handled_remove_error = self._sanitize_dnf_error_msg_remove(pkg_spec, to_native(e))
-                            if is_failure:
-                                failure_response['failures'].append('{0} - {1}'.format(pkg_spec, to_native(e)))
-                            else:
-                                response['results'].append(handled_remove_error)
-                        continue
-
-                    installed_pkg = dnf.subject.Subject(pkg_spec).get_best_query(
-                        sack=self.base.sack).installed().run()
-
-                    for pkg in installed_pkg:
-                        self.base.remove(str(pkg))
+                    try:
+                        self.base.remove(pkg_spec)
+                    except dnf.exceptions.MarkingError as e:
```

Comment:
This looks fine, we are actually using a subclass `dnf.exceptions.PackagesNotInstalledError` if you want to be more explicit here, but you can keep it like it is.

Reply:
Hmm, yeah, rather stay with the stable API class. I haven't noticed the API markup is missing there.